contracts/index/IReserveVault.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BUSDL-1.1
+pragma solidity 0.6.11;
+
+/**
+ * @notice For vaults that keep a separate reserve of tokens
+ */
+interface IReserveVault {
+    /**
+     * @notice Log when the percent held in reserve is changed
+     * @param reservePercentage The new percent held in reserve
+     */
+    event ReservePercentageChanged(uint256 reservePercentage);
+
+    /**
+     * @notice Set a new percent of tokens to hold in reserve
+     * @param reservePercentage_ The new percent
+     */
+    function setReservePercentage(uint256 reservePercentage_) external;
+
+    /**
+     * @notice Transfer an amount of tokens to the LP Account
+     * @dev This should only be callable by the `LpAccountFunder`
+     * @param amount The amount of tokens
+     */
+    function transferToLpAccount(uint256 amount) external;
+
+    /**
+     * @notice Get the amount of tokens missing from the reserve
+     * @dev A negative amount indicates extra tokens not needed for the reserve
+     * @return The amount of missing tokens
+     */
+    function getReserveTopUpValue() external view returns (int256);
```

Comment:
Let's come up with a new name for this while we have the opportunity. It may make sense to switch the sign and make it a `uint256` as well, because it will only be used to determine excess tokens, never tokens that should be returned.

Maybe we should overall avoid the reference to the LP Account, even with `transferToLpAccount`. All the vault fundamentally cares about is that there is something it can transfer to, that has permission to initiate the transfer, and that it can only transfer the excess tokens.

One way we could think of this is as a 0% interest loan to another contract. Might be the wrong approach, but let's see how that might look:
```
function lendUnderlyer(uint256 amount) external;

function getMaxUnderlyerLoan() external view returns (uint256);
```

Could also keep it simpler and stay with the transfer terminology:
```
function transferUnderlyer(uint256 amount) external;

function getTransferUnderlyerMaxAmount() external view returns (uint256);
```

In both cases I think it makes sense to drop the sign and invert the calculation so excess tokens is a positive number. Should simplify the rest of any calculations.